fix: update status of Email Campaign

harshkhatri7287 opened the pull request:
The status of the email campaign is scheduled to update daily, but it was not updating it in the database,
Now the updated status is directly set in the database also. 
Issue #39179

harshkhatri7287 committed:
Email campaign's status is being updated in database

khushrrp commented on erpnext/crm/doctype/email_campaign/email_campaign.py:
```diff
@@ -79,10 +79,13 @@ def update_status(self):
 		today_date = getdate(today())
 		if start_date > today_date:
 			self.status = "Scheduled"
+			frappe.db.set_value("Email Campaign", self.name, "status", "Scheduled")
```
use self.status in place of all status values here.

harshkhatri7287 committed:
Status update for Email Campaigns

Initially update status function for Email Campaign were running daily but not setting the values in database, Now the status is being updated in database.

harshkhatri7287 commented:
@leela @venetanji Please proceed this PR for pending tests.

khushali-beehyv commented:
@deepeshgarg007 @rohitwaghchaure  Can you please help this PR to proceed for the tests and the review process ?

barredterra reviewed (changes requested):
Thanks for the fix! I'm afraid this is not the right place for writing to the database.

Better add `email_campaign.save()` here, after line 146:

https://github.com/frappe/erpnext/blob/22291d32e6dfee445b15a2519e111a781b73abd5/erpnext/crm/doctype/email_campaign/email_campaign.py#L142-L146

harshkhatri7287 committed:
Update status of email campaign

Issue: update_status function from hooks was running daily but not updating the email campaign status.
Fix: Saved the email campaign object after changing it.

barredterra committed:
Merge remote-tracking branch 'upstream/develop' into bug39179

harshkhatri7287 commented:
Hi @barredterra, One required test case is failing, Can you please explain this, why is it happening?
